ls1intum/Artemis
# `General`: Improve CSV header parsing when importing users into course groups

### Checklist
#### General


#### Client


### Motivation and Context


This is a follow-up PR to the fix [b054013](https://github.com/ls1intum/Artemis/commit/b054013ff172ee355d41e1fe6c11e474cffaa62e), which closed #4939.
The changes introduced in this PR simplify the code and also make it clearer where new header fields need to be added in the future.
In addition, this PR addresses an additional improvement mentioned in a comment in [#4939](https://github.com/ls1intum/Artemis/issues/4939#issuecomment-1106570075): 
> All "not found" lines should be red, not only every second one
This refers to the text color.

### Description

This PR transitions from a single object, which previously stored the possible csv header fields in the object's parameters, to several arrays.

### Steps for Testing

Prerequisites:
- 1 Course

1. Log in to Artemis as Admin
2. Navigate to Course Administration
3. Select a course and go to the student overview
4. Click Import Users
5. Select the CSV file

- [invalid_rows_2_and_6.csv](https://github.com/ls1intum/Artemis/files/8543832/invalid_rows_2_and_6.csv)
- [valid_test_3_of_5_users_imported.csv](https://github.com/ls1intum/Artemis/files/8543825/valid_test_3_of_5_users_imported.csv)

7. Click on Import
- for the invalid file an error message should be shown, indicating errors in row 2 and 6:
![Bildschirmfoto 2022-04-22 um 22 26 55](https://user-images.githubusercontent.com/9954674/164789012-419daeb5-35ab-4fe1-8501-0911642fc150.png)

- for the valid file 3 out of 5 users should get imported



### Review Progress





#### Code Review
#### Manual Tests

### Test Coverage


| Class/File | Branch | Line |
|------------|-------:|-----:|
| users-import-dialog.component.ts | 78.3% | 89.52% |


### Screenshots



Before the changes, only every second line the import status was applied:
![Bildschirmfoto 2022-04-22 um 22 15 08](https://user-images.githubusercontent.com/9954674/164788397-b9d61223-fb7c-497d-87b5-fe58529b44c3.png)

After the changes the text color corresponding to the import status is applied correctly regardless of the row position:
![Bildschirmfoto 2022-04-22 um 22 18 43](https://user-images.githubusercontent.com/9954674/164788380-529bea3f-9583-4b5c-a901-0d69c27dd3b1.png)
